api/v1alpha1/velero_types.go

```diff
@@ -109,44 +132,40 @@ type VeleroSpec struct {
 	// https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config
 	// +optional
 	PodDnsConfig corev1.PodDNSConfig `json:"podDnsConfig,omitempty"`
-	// RestoreResourceVersionPriority represents a configmap that will be created if defined for use in conjunction with `EnableAPIGroupVersions` feature flag
-	// Defining this field automatically add EnableAPIGroupVersions to the velero server feature flag
-	// +optional
-	RestoreResourcesVersionPriority string `json:"restoreResourcesVersionPriority,omitempty"`
 	// BackupImages is used to specify whether you want to deploy a registry for enabling backup and restore of images
 	// +optional
 	BackupImages *bool `json:"backupImages,omitempty"`
-	// If you need to install Velero without a default backup storage location NoDefaultBackupLocation flag is required for confirmation
-	// +optional
-	NoDefaultBackupLocation bool `json:"noDefaultBackupLocation,omitempty"`
+
+	// ServerConfiguration is used to configure the backup application's server config
+	ServerConfiguration *ServerConfig `json:"serverConfiguration"`
 }
 
 // VeleroStatus defines the observed state of Velero
-type VeleroStatus struct {
+type BackupApplicationStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
 //+kubebuilder:object:root=true
 //+kubebuilder:subresource:status
 
```

Comment:
I wonder if we should add `shortName=...`, `printerColumn` to the list of kube builder annotations before GA. Having some crude validation a the API level will only be great https://book.kubebuilder.io/reference/generating-crd.html#validation